src/components/Sidebar/ActionNode.tsx

```diff
@@ -1,49 +1,212 @@
+  { label: 'Service Request', value: 'Service Request' },
+  { label: 'Careplan', value: 'Careplan' }
+];
+
+const codeSystemOptions = [
+  { label: 'RXNORM', value: 'http://www.nlm.nih.gov/research/umls/rxnorm' },
+  { label: 'SNOMED', value: 'http://snomed.info/sct' },
+  { label: 'LOINC', value: ' http://loinc.org' }
+];
+
 interface ActionNodeProps {
   pathway: Pathway;
-  currentNode: State;
+  currentNode: GuidanceState;
   changeNodeType: (event: string) => void;
   addNode: (event: string) => void;
+  updatePathway: (pathway: Pathway) => void;
 }
 
-const ActionNode: FC<ActionNodeProps> = ({ pathway, currentNode, changeNodeType, addNode }) => {
+const ActionNode: FC<ActionNodeProps> = ({
+  pathway,
+  currentNode,
+  changeNodeType,
+  addNode,
+  updatePathway
+}) => {
   const styles = useStyles();
   const selectNodeType = useCallback(
     (event: ChangeEvent<{ value: string }>): void => {
       changeNodeType(event?.target.value || '');
     },
     [changeNodeType]
   );
+  const currentNodeKey = currentNode?.key;
+
+  const changeCode = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+
+      const code = event?.target.value || '';
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      const action: Action = currentState.action[0];
+      action.resource.code.coding[0].code = code;
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
+
+  const changeTitle = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+
+      const title = event?.target.value || '';
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      const action: Action = currentState.action[0];
+      action.resource.title = title;
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
 
+  const selectActionType = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+      const value = event?.target.value || '';
+      const actionType = actionTypeOptions.find(option => {
+        return option.value === value;
+      });
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      let action: Action;
+      if (actionType?.label === 'Careplan') {
+        action = {
+          type: 'create',
+          description: '',
+          resource: {
+            id: shortid.generate(),
+            resourceType: actionType?.label,
+            title: ''
+          }
+        };
+      } else if (
+        currentState.action.length > 0 &&
+        currentState.action[0].resource.resourceType !== 'Careplan'
+      ) {
+        action = currentState.action[0];
+        action.resource.resourceType = actionType?.label;
```

Comment:
Did the CarePlan replace the need for multiple actions? If so then we should probably update the data model to make Action a single object instead of a list.
The reason I bring this up is because it looks like this block means we cannot add multiple actions - it just replaces the action type of the first one

Reply:
I told Keeyan separately that since we only ever show one action at a time, for starters this PR should use the existing data model and only put one item in the list. If we want to change the data model, which I would be good with, let's do that separately.

Reply:
I'm not sure what to do here since the designs only show one action being added but the data model allows for multiple.  I've talked to dylan and we decided to keep the data model the same even though it only allows 1 action for now.